### PR TITLE
Sphinx Docs

### LINKED ISSUE #55: Add Sphinx
Pull request converts from the existing README here 

https://github.com/ga4gh/data-object-service-schemas/pull/53

Review/merge and commit as a basis.

### PR DESCRIPTION
**Pull request by david4096**
Close #55 

First pass at adding sphinx docs. Exposed that docstrings for many functions were still needed.

Considering this mergeable and can add further docstrings to the code linking back as necessary.

More work is needed on smoothing out the README, sphinx docs will help it stay short. Also need to add the development and contributing docs.

### CODE REVIEW

File: docs/source/intro.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+------------------------------
+
+You can use ``pip`` to install a python client and server that
+implements these schemas.
+
+::
+
+    virtualenv env
+    source env/bin/activate
+    pip install git+git://github.com/ga4gh/data-object-service-schemas@master --process-dependency-links
```
Comment (natanlao) [inline docs/source/intro.rst]: Is there a reason this isn't `pip install ga4gh-dos-schemas `?

Comment (david4096) [reply docs/source/intro.rst]: Newp, just hadn't published when this was made I believe. +1 to deleting/moving a lot of cruft on this readme and replacing with "pip install" and a client demo.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <img src="https://www.ga4gh.org/gfx/GA-logo-horizontal-tag-RGB.svg" alt="Drawing" style="width: 120px;" alt="GA4GH colored ring logo"/>
 
-![travis status](https://travis-ci.org/ga4gh/data-object-service-schemas.svg?branch=master)
+[![Build Status](https://travis-ci.org/ga4gh/data-object-service-schemas.svg?branch=master)](https://travis-ci.org/ga4gh/data-object-service-schemas)
+[![Swagger syntax](https://online.swagger.io/validator?url=https://raw.githubusercontent.com/ga4gh/data-object-service-schemas/master/openapi/data_object_service.swagger.yaml)](https://github.com/swagger-api/validator-badge)
```
Comment (david4096) [inline README.md]: Nice!

File: setup.py [hunk 3]
```diff
@@ -64,6 +64,7 @@
     },
     package_dir={'': 'python'},
     long_description=long_description,
+    long_desceiption_content_type='text/markdown',
```
Comment (david4096) [inline setup.py]: Typo in `description`... I did not know you could do this!

**Comment by david4096** [comment]
Thanks a bunch @natanlao! Removed the link to RTD for now and some small changes I was able to catch (an old link and poor wording in a comment).

**Pull request merged**